app/src/main/java/com/ozonehis/eip/Utils.java

```diff
@@ -0,0 +1,34 @@
+package com.ozonehis.eip;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Utils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Utils.class);
+
+    private static boolean shuttingDown = false;
+
+    public static void setShuttingDown() {
+        shuttingDown = true;
+        LOGGER.info("Received application shutting down event");
+    }
+
+    public static boolean isShuttingDown() {
+        return shuttingDown;
+    }
+
+    public static synchronized void shutdown() {
+        if (isShuttingDown()) {
+            LOGGER.info("Application is already shutting down");
+            return;
+        }
+
+        setShuttingDown();
+
+        LOGGER.info("Shutting down the application...");
+
+        // Shutdown in a new thread to ensure other background shutdown threads complete too
+        new Thread(() -> System.exit(129)).start();
```

Comment:
Shouldn't the exit status be `0`? Most tooling will interpret this as a failure.

Reply:
Took inspiration from here https://github.com/openmrs/openmrs-eip/blob/master/commons/src/main/java/org/openmrs/eip/Utils.java#L107

Reply:
Makes sense! However, that `Utils#shutdown()` is only called in the context of a fatal error having occurred, e.g., [here](https://github.com/openmrs/openmrs-eip/blob/e6123691e6ece98ec76592dd25f11057fc717735/openmrs-watcher/src/main/java/org/openmrs/eip/mysql/watcher/FailureTolerantMySqlConnector.java#L57), [here](https://github.com/openmrs/openmrs-eip/blob/e6123691e6ece98ec76592dd25f11057fc717735/openmrs-watcher/src/main/java/org/openmrs/eip/mysql/watcher/FailureTolerantMySqlConnector.java#L76), or [here](https://github.com/openmrs/openmrs-eip/blob/e6123691e6ece98ec76592dd25f11057fc717735/openmrs-watcher/src/main/java/org/openmrs/eip/mysql/watcher/route/WatcherShutdownRoute.java#L22), so it makes sense there to exit with a code indicating an error. Here, however, the shutdown is a normal operation, just stopping the process, so `0`'s a better code to use.

Reply:
Understood.